core/trino-main/src/main/java/io/trino/server/security/ServerSecurityModule.java

```diff
@@ -62,6 +62,7 @@ protected void setup(Binder binder)
                 .internalOnlyResource(StoreResource.class);
 
         binder.bind(PasswordAuthenticatorManager.class).in(Scopes.SINGLETON);
+        configBinder(binder).bindConfig(PasswordAuthenticatorConfig.class);
```

Comment:
It should not be possible to specify password config properties when `password` authentication is not used.

Maybe `PasswordAuthenticatorManager` (along with config) can be binded conditionally? Add `PasswordAuthenticatorSupportModule` as for Jwt and Oauth2.

Reply:
You can use https://google.github.io/guice/api-docs/4.1/javadoc/com/google/inject/multibindings/OptionalBinder.html